fdbserver/BlobWorker.actor.cpp

```diff
@@ -2192,6 +2207,26 @@ ACTOR Future<Void> doBlobGranuleFileRequest(Reference<BlobWorkerData> bwData, Bl
 		}
 	}
 
+	state Optional<Key> tenantPrefix;
+	if (req.tenantInfo.name.present()) {
+		ASSERT(req.tenantInfo.tenantId != TenantInfo::INVALID_TENANT);
+		auto tenantEntry = bwData->tenantInfoById.find(req.tenantInfo.tenantId);
+		if (tenantEntry != bwData->tenantInfoById.end()) {
+			ASSERT(tenantEntry->second.id == req.tenantInfo.tenantId);
+			tenantPrefix = tenantEntry->second.prefix;
+		} else {
+			// FIXME - better way. Wait on retry here, or just have better model for tenant metadata?
+			// Just throw wrong_shard_server and make the client retry and assume we load it later
+			TraceEvent(SevDebug, "BlobWorkerRequestUnknownTenant", bwData->id)
+			    .suppressFor(5.0)
+			    .detail("TenantName", req.tenantInfo.name.get())
+			    .detail("TenantId", req.tenantInfo.tenantId);
+			throw wrong_shard_server();
```

Comment:
I had been using the `unknown_tenant` error for a server to signal to the client that it wasn't aware of the tenant. It in many ways behaves similarly to `wrong_shard_server`.